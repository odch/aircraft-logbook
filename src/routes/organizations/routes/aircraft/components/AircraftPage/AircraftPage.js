import React from 'react'
import PropTypes from 'prop-types'
import Header from '../../../../../../containers/HeaderContainer'
import AircraftDetail from '../../containers/AircraftDetailContainer'

const AircraftPage = ({ router: { match } }) => (
  <React.Fragment>
    <Header />
    <AircraftDetail match={match} />
  </React.Fragment>
)

AircraftPage.propTypes = {
  router: PropTypes.shape({
    match: PropTypes.shape({
      params: PropTypes.object.isRequired
    }).isRequired
  })
}

export default AircraftPage
