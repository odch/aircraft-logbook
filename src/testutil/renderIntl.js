import React from 'react'
import renderer from 'react-test-renderer'
import { createMount } from '@material-ui/core/test-utils'
import { IntlProvider } from 'react-intl'
import pretty from 'pretty'
import messages from '../messages'

const LOCALE = 'de'

const renderInIntlProvider = component => (
  <IntlProvider locale={LOCALE} messages={messages[LOCALE]}>
    {component}
  </IntlProvider>
)

const renderIntl = (component, options) =>
  renderer.create(renderInIntlProvider(component), options)

export const renderIntlUpdate = (renderer, component, options) =>
  renderer.update(renderInIntlProvider(component), options)

export const renderIntlMaterial = (component, returnPrettyHtml = false) => {
  const rendered = createMount()(renderInIntlProvider(component))
  return returnPrettyHtml ? pretty(rendered.html()) : rendered
}

export default renderIntl
