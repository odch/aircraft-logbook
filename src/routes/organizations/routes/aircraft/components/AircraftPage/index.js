import AircraftPage from './AircraftPage'

export default AircraftPage
