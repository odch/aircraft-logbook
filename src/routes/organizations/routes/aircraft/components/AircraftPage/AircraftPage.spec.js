import React from 'react'
import { Provider } from 'react-redux'
import { BrowserRouter as Router } from 'react-router-dom'
import configureStore from 'redux-mock-store'
import renderIntl from '../../../../../../testutil/renderIntl'
import AircraftPage from './AircraftPage'

describe('routes', () => {
  describe('organizations', () => {
    describe('routes', () => {
      describe('aircraft', () => {
        describe('components', () => {
          describe('AircraftPage', () => {
            it('renders correctly', () => {
              const state = {
                firebase: {
                  auth: {
                    isEmpty: false
                  },
                  profile: {}
                },
                main: {
                  app: {
                    organizations: [{ id: 'my_org' }]
                  }
                },
                firestore: {
                  ordered: {
                    'fligts-o7flC7jw8jmkOfWo8oyA': [
                      {
                        id: 'sStfyLd2XArT7oUZPFDn',
                        departureAerodrome: { name: 'Lommis' },
                        destinationAerodrome: { name: 'Lommis' },
                        blockOffTime: {
                          toDate: () => Date.parse('2018-11-20 10:00')
                        },
                        blockOnTime: {
                          toDate: () => Date.parse('2018-11-20 11:10')
                        },
                        member: {
                          firstname: 'Max',
                          lastname: 'Muster'
                        }
                      },
                      {
                        id: 'sStfyLd2XArT7oUZPFDa',
                        departureAerodrome: { name: 'Lommis' },
                        destinationAerodrome: { name: 'Lommis' },
                        blockOffTime: {
                          toDate: () => Date.parse('2018-11-21 10:00')
                        },
                        blockOnTime: {
                          toDate: () => Date.parse('2018-11-21 11:10')
                        },
                        member: {
                          firstname: 'Hans',
                          lastname: 'Meier'
                        }
                      }
                    ]
                  },
                  data: {
                    organizationAircrafts: {
                      o7flC7jw8jmkOfWo8oyA: {
                        id: 'o7flC7jw8jmkOfWo8oyA',
                        registration: 'HBKFW'
                      },
                      BKi7HYAIoe1i75H3LMk1: {
                        id: 'BKi7HYAIoe1i75H3LMk1',
                        registration: 'HBKOF'
                      }
                    }
                  }
                }
              }

              const store = configureStore()(state)

              const props = {
                router: {
                  match: {
                    params: {
                      organizationId: 'my_org'
                    }
                  }
                }
              }

              const tree = renderIntl(
                <Provider store={store}>
                  <Router>
                    <AircraftPage {...props} />
                  </Router>
                </Provider>
              ).toJSON()
              expect(tree).toMatchSnapshot()
            })
          })
        })
      })
    })
  })
})
