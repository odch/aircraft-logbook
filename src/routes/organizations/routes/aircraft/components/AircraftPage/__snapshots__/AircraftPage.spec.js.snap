// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`routes organizations routes aircraft components AircraftPage renders correctly 1`] = `
Array [
  <header
    className="MuiPaper-root-14 MuiPaper-elevation4-20 MuiAppBar-root-5 MuiAppBar-positionStatic-9 MuiAppBar-colorPrimary-12"
  >
    <div
      className="MuiToolbar-root-41 MuiToolbar-regular-43 MuiToolbar-gutters-42"
    >
      <h2
        className="MuiTypography-root-45 MuiTypography-title-51 MuiTypography-colorInherit-65 Header-grow-2"
      >
        <a
          className="Header-homeLink-4"
          href="/"
          onClick={[Function]}
        >
          Open Digital Flugreisebuch
        </a>
      </h2>
      <div>
        <button
          aria-haspopup="true"
          aria-owns={null}
          className="MuiButtonBase-root-77 MuiIconButton-root-71 MuiIconButton-colorInherit-72"
          data-cy="user-button"
          disabled={false}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onKeyDown={[Function]}
          onKeyUp={[Function]}
          onMouseDown={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
          onTouchEnd={[Function]}
          onTouchMove={[Function]}
          onTouchStart={[Function]}
          tabIndex="0"
          type="button"
        >
          <span
            className="MuiIconButton-label-76"
          >
            <svg
              aria-hidden="true"
              className="MuiSvgIcon-root-80"
              focusable="false"
              role="presentation"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm0 3c1.66 0 3 1.34 3 3s-1.34 3-3 3-3-1.34-3-3 1.34-3 3-3zm0 14.2c-2.5 0-4.71-1.28-6-3.22.03-1.99 4-3.08 6-3.08 1.99 0 5.97 1.09 6 3.08-1.29 1.94-3.5 3.22-6 3.22z"
              />
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
            </svg>
          </span>
          <span
            className="MuiTouchRipple-root-89"
          />
        </button>
      </div>
    </div>
  </header>,
  <div
    className="LoadingIcon-container-101"
  >
    <div
      className="MuiCircularProgress-root-103 MuiCircularProgress-colorPrimary-106 MuiCircularProgress-indeterminate-105 LoadingIcon-icon-102"
      role="progressbar"
      style={
        Object {
          "height": 40,
          "width": 40,
        }
      }
    >
      <svg
        className="MuiCircularProgress-svg-108"
        viewBox="22 22 44 44"
      >
        <circle
          className="MuiCircularProgress-circle-109 MuiCircularProgress-circleIndeterminate-111"
          cx={44}
          cy={44}
          fill="none"
          r={20.2}
          strokeWidth={3.6}
          style={Object {}}
        />
      </svg>
    </div>
  </div>,
]
`;
